aws/network-primitives/network.tf

```diff
@@ -0,0 +1,193 @@
+resource "aws_vpc" "gemini_vpc" {
+  cidr_block           = "172.31.0.0/16"
+  enable_dns_support   = true
+  enable_dns_hostnames = true
+
+  tags = {
+    name = "${var.network_name}-vpc"
+  }
+}
+
+resource "aws_subnet" "public_subnets" {
+  count                   = length(var.public_subnet_cidrs)
+  vpc_id                  = aws_vpc.gemini_vpc.id
+  cidr_block              = element(var.public_subnet_cidrs, count.index)
+  availability_zone       = element(var.azs, count.index)
+  map_public_ip_on_launch = "true"
+
+  tags = {
+    Name = "${var.network_name}-public-subnet-${count.index}"
+  }
+}
+
+resource "aws_subnet" "private_subnets" {
+  count             = length(var.private_subnet_cidrs)
+  vpc_id            = aws_vpc.gemini_vpc.id
+  cidr_block        = element(var.private_subnet_cidrs, count.index)
+  availability_zone = element(var.azs, count.index)
+
+  tags = {
+    Name = "${var.network_name}-private-subnet-${count.index}"
+  }
+
+  lifecycle {
+    prevent_destroy = false
+  }
+
+}
+
+
+resource "aws_internet_gateway" "gw" {
+  count  = length(var.public_subnet_cidrs)
+  vpc_id = aws_vpc.gemini_vpc.id
+
+  tags = {
+    Name = "${var.network_name}-igw-public-subnet-${count.index}"
+  }
+
+  lifecycle {
+    prevent_destroy = false
+  }
+}
+
+resource "aws_route_table" "public_route_table" {
+  count  = length(var.public_subnet_cidrs)
+  vpc_id = aws_vpc.gemini_vpc.id
+
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.gw[count.index].id
+  }
+
+  route {
+    ipv6_cidr_block = "::/0"
+    gateway_id      = aws_internet_gateway.gw[count.index].id
+  }
+
+  tags = {
+    Name = "${var.network_name}-public-route-tbl-${count.index}"
+  }
+
+  depends_on = [
+    aws_internet_gateway.gw
+  ]
+}
+
+resource "aws_route_table_association" "public_route_table_subnets_association" {
+  count          = length(var.public_subnet_cidrs)
+  subnet_id      = element(aws_subnet.public_subnets.*.id, count.index)
+  route_table_id = element(aws_route_table.public_route_table.*.id, count.index)
+}
+
+
+resource "aws_route_table" "private_route_table" {
+  count  = length(var.private_subnet_cidrs)
+  vpc_id = aws_vpc.gemini_vpc.id
+
+  route {
+    cidr_block     = "0.0.0.0/0"
+    nat_gateway_id = aws_nat_gateway.nat_gateway[count.index].id
+  }
+
+  tags = {
+    Name = "${var.network_name}-private-route-tbl-${count.index}"
+  }
+}
+
+resource "aws_route_table_association" "private_route_table_subnets_association" {
+  count          = length(var.private_subnet_cidrs)
+  subnet_id      = element(aws_subnet.private_subnets.*.id, count.index)
+  route_table_id = element(aws_route_table.private_route_table.*.id, count.index)
+}
+
+resource "aws_security_group" "gemini_sg" {
```

Comment:
Looks like port 50000 is missing in this list despite used in DSN bootstrap node container

Reply:
Will add this, actually now i think of it that port was never set in firewall from our DO declarations.